src/generic/Matrix.jl

```diff
@@ -148,63 +148,44 @@ end
 #
 ###############################################################################
 
+# create a zero matrix
 function (a::MatSpace{T})() where {T <: NCRingElement}
-   R = base_ring(a)
-   entries = Matrix{T}(undef, a.nrows, a.ncols)
-   for i = 1:a.nrows
-      for j = 1:a.ncols
-         entries[i, j] = zero(R)
-      end
-   end
-   z = MatSpaceElem{T}(R, entries)
-   return z
+   return zero_matrix(base_ring(a), nrows(a), ncols(a))::dense_matrix_type(T)
```

Comment:
So I was worried earlier: what if there is more than one "relevant" matrix type per ring?

One way to solve this would be to add a second type parameter to MatSpace indicating the matrix type, i.e.:
```
struct MatSpace{T <: NCRingElement, MatT}
```
then we can have a constructor which, `MatT` is omitted, sets it to `dense_matrix_type(T)`. Also `matrix_space(ring,r,c)` will use `dense_matrix_type(typeof(ring))` by default.

Then we can do
```
elem_type(::Type{MatSpace{T,MatT}}) where {T,MatT} = MatT
```
Finally, we need a clean way to instantiate a zero matrix of a given type (and base ring). We mandate that this must always work:
```
  new_zero_mat  =  MatT(ring, r, c)
```
and then we can redefine e.g. this function as
```
function (a::MatSpace{T,MatT})() where {T <: NCRingElement, MatT}
  MatT(base_ring(a), nrows(a), ncols(a))
end
```